F# compiler crashes with SIGILL (exit 132) on Linux ARM64

## Summary

In the .NET 11 SDK build `11.0.100-preview.6.26304.123`, invoking the F# compiler on **Linux ARM64** (Ubuntu 24.04) consistently crashes the spawned `dotnet` process with **exit code 132 (SIGILL — illegal instruction)**. Every template type that compiles F# code is affected. C# and VB compilation on the same SDK/host succeed.

## Repro

On Ubuntu 24.04 ARM64 with SDK `11.0.100-preview.6.26304.123` installed at `/usr/share/dotnet/`:

```bash
dotnet new console -lang F# -o repro
cd repro
dotnet build
```

## Observed error

```
/usr/share/dotnet/sdk/11.0.100-preview.6.26304.123/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(383,5): message NETSDK1057: You are using a preview version of .NET.
/usr/share/dotnet/sdk/11.0.100-preview.6.26304.123/FSharp/Microsoft.FSharp.Targets(354,9): error MSB6006: "dotnet" exited with code 132.

Build FAILED.
    0 Warning(s)
    1 Error(s)
```

`Microsoft.FSharp.Targets:354` is the `Fsc` task invocation. Exit code 132 on Linux = `128 + SIGILL(4)`, indicating the spawned compiler process executed an illegal CPU instruction.

## Scope

- **Affected:** Linux ARM64 (Ubuntu 24.04 verified; cascades into `LinuxInstallerTests.DebScenarioTest` / `RpmScenarioTest` for `runtime-deps:10.0-noble` and `runtime-deps:10.0-azurelinux3.0` ARM64 images)
- **Not affected:** Linux x64 (same SDK, same scenario tests pass)
- **Not affected:** C# / VB compilation on the same ARM64 host with the same SDK
- **Affected templates** (all F# variants): `console`, `classlib`, `web`, `webapi`, `mvc`, `xunit`, `nunit`, `mstest`, plus complex/multi-TFM variants
- **Affected SDK flavors:** Both Microsoft-built SDK (`ValidateInstallers_Linux_arm64`) and source-built SDK (`SB_Ubuntu2404Arm64_Offline_MsftSdk_Validation_arm64`) — implicating a shared component, not a build-pipeline difference

## Evidence

- Failing VMR build (latest): https://dev.azure.com/dnceng/internal/_build/results?buildId=2992792
  - VMR SHA: `cc468399`
  - Branch: `refs/heads/main`
  - Failing jobs:
    - `ValidateInstallers_Linux_arm64` (stage: VMR Validation)
    - `SB_Ubuntu2404Arm64_Offline_MsftSdk_Validation_arm64` (stage: VMR Source-Only Validation)

--- a/src/coreclr/vm/arm64/asmhelpers.S
+++ b/src/coreclr/vm/arm64/asmhelpers.S
@@ -160,11 +160,16 @@ NESTED_END OnHijackTripThread, _TEXT
 // void* PacStripPtr(void *);
 // This function strips the pointer of PAC info that is passed as an argument.
 // We prefer to strip a pointer where it's not going to be used to branch execution to.
+// It is a no-op on non-PAC enabled machines.
 .arch_extension pauth
-    LEAF_ENTRY PacStripPtr, _TEXT
-        xpaci x0
-        ret
-    LEAF_END PacStripPtr, _TEXT
+    NESTED_ENTRY PacStripPtr, _TEXT, NoHandler
+        PROLOG_SAVE_REG_PAIR_INDEXED fp, lr, -16
+        mov lr, x0
+        xpaclri
+        mov x0, lr
+        EPILOG_RESTORE_REG_PAIR_INDEXED fp, lr, 16
+        EPILOG_RETURN
+    NESTED_END PacStripPtr, _TEXT
 
 // void* PacSignPtr(void *, void *);
 // This function signs the input pointer using x1 as salt. It is a no-op on non-PAC enabled machines.

--- a/src/coreclr/vm/arm64/asmhelpers.asm
+++ b/src/coreclr/vm/arm64/asmhelpers.asm
@@ -323,10 +323,15 @@ OnHijackTripThreadReturn
 ; void* PacStripPtr(void *);
 ; This function strips the pointer of PAC info that is passed as an argument.
 ; We prefer to strip a pointer where it's not going to be used to branch execution to.
-    LEAF_ENTRY PacStripPtr
-        DCD     0xDAC143E0  ; xpaci x0 instruction in binary to avoid requiring PAC-enabled assemblers
-        ret
-    LEAF_END PacStripPtr
+; It is a no-op on non-PAC enabled machines.
+    NESTED_ENTRY PacStripPtr
+        PROLOG_SAVE_REG_PAIR_INDEXED fp, lr, #-16!
+        mov lr, x0
+        DCD     0xD50320FF  ; xpaclri instruction in binary to avoid requiring PAC-enabled assemblers
+        mov x0, lr
+        EPILOG_RESTORE_REG_PAIR_INDEXED fp, lr, 16
+        EPILOG_RETURN
+    NESTED_END
 
 ; void* PacSignPtr(void *, void *);
 ; This function signs the input pointer using x1 as salt. It is a no-op on non-PAC enabled machines.

--- a/src/coreclr/vm/arm64/asmhelpers.asm
+++ b/src/coreclr/vm/arm64/asmhelpers.asm
@@ -325,11 +325,11 @@ OnHijackTripThreadReturn
 ; We prefer to strip a pointer where it's not going to be used to branch execution to.
 ; It is a no-op on non-PAC enabled machines.
     NESTED_ENTRY PacStripPtr
-        PROLOG_SAVE_REG_PAIR_INDEXED fp, lr, #-16!
+        PROLOG_SAVE_REG_PAIR fp, lr, #-16!
         mov lr, x0
         DCD     0xD50320FF  ; xpaclri instruction in binary to avoid requiring PAC-enabled assemblers
         mov x0, lr
-        EPILOG_RESTORE_REG_PAIR_INDEXED fp, lr, 16
+        EPILOG_RESTORE_REG_PAIR fp, lr, #16!
         EPILOG_RETURN
     NESTED_END
 

